src/main/java/org/example/librarymanagement/entity/Account.java

```diff
@@ -0,0 +1,80 @@
+package org.example.librarymanagement.entity;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.example.librarymanagement.enumeration.AccountStatus;
+import org.example.librarymanagement.enumeration.Role;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.List;
+
+@Getter
+@Setter
+@Inheritance(strategy = InheritanceType.JOINED)
+@NoArgsConstructor
+@MappedSuperclass
+public class Account{
```

Comment:
Em tạo 1 abstract class handle những thông tin audit, và implement event listener để handle set data cho audit nha
==> Toàn bộ entity sẽ extend class audit này